contracts/protocol/facets/SequentialCommitHandlerFacet.sol

```diff
@@ -91,44 +92,44 @@ contract SequentialCommitHandlerFacet is IBosonSequentialCommitHandler, PriceDis
         }
 
         if (_priceDiscovery.side == Side.Bid) {
+            // @TODO why don't allow third party to call?
             require(seller == msgSender(), NOT_VOUCHER_HOLDER);
         }
 
         // First call price discovery and get actual price
         // It might be lower tha submitted for buy orders and higher for sell orders
-        uint256 actualPrice = fulFilOrder(_exchangeId, tokenAddress, _priceDiscovery, _buyer, offer.sellerId);
+        uint256 actualPrice = fulfilOrder(_tokenId, offer, _priceDiscovery, _buyer);
 
         // Calculate the amount to be kept in escrow
         uint256 escrowAmount;
+
         {
             // Get sequential commits for this exchange
-            SequentialCommit[] storage sequentialCommits = protocolEntities().sequentialCommits[_exchangeId];
+            ExchangeCosts[] storage exchangeCosts = protocolEntities().exchangeCosts[exchangeId];
 
             {
                 // Calculate fees
-                uint256 protocolFeeAmount = tokenAddress == protocolAddresses().token
-                    ? protocolFees().flatBoson
-                    : (protocolFees().percentage * actualPrice) / 10000;
+                uint256 protocolFeeAmount = getProtocolFee(tokenAddress, actualPrice);
 
                 // Calculate royalties
                 (, uint256 royaltyAmount) = IBosonVoucher(protocolLookups().cloneAddress[offer.sellerId]).royaltyInfo(
-                    _exchangeId,
+                    exchangeId,
                     actualPrice
                 );
 
                 // Verify that fees and royalties are not higher than the price.
                 require((protocolFeeAmount + royaltyAmount) <= actualPrice, FEE_AMOUNT_TOO_HIGH);
 
                 // Get price paid by current buyer
-                uint256 len = sequentialCommits.length;
-                uint256 currentPrice = len == 0 ? offer.price : sequentialCommits[len - 1].price;
+                uint256 len = exchangeCosts.length;
+                uint256 currentPrice = len == 0 ? offer.price : exchangeCosts[len - 1].price;
 
                 // Calculate the minimal amount to be kept in the escrow
                 escrowAmount = Math.max(actualPrice, protocolFeeAmount + royaltyAmount + currentPrice) - currentPrice;
 
                 // Update sequential commit
-                sequentialCommits.push(
-                    SequentialCommit({
+                exchangeCosts.push(
```

Comment:
maybe comment above should be different?

Reply:
Changed.